wrapper/src/main/java/software/amazon/jdbc/ConnectionPluginChainBuilder.java

```diff
@@ -110,6 +112,7 @@ public class ConnectionPluginChainBuilder {
           put(ConnectTimeConnectionPluginFactory.class, WEIGHT_RELATIVE_TO_PRIOR_PLUGIN);
           put(ExecutionTimeConnectionPluginFactory.class, WEIGHT_RELATIVE_TO_PRIOR_PLUGIN);
           put(DeveloperConnectionPluginFactory.class, WEIGHT_RELATIVE_TO_PRIOR_PLUGIN);
+          put(CustomEndpointPluginFactory.class, WEIGHT_RELATIVE_TO_PRIOR_PLUGIN);
```

Comment:
I'd put it before RW splitting plugin.

Reply:
If we put it before the RW splitting plugin the monitor thread will actually be created later rather than sooner, because the monitoring thread is created after connecting when we are returning backwards up the connect pipeline.

Reply:
We usually start monitoring before opening a connection. But I agree with your point that it makes more sense to start it when connection is successfully established. Either way work.